nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/clients/OAuth2ClientServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (C) Copyright 2014-2017 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Arnaud Kervern
+ */
+package org.nuxeo.ecm.platform.oauth2.clients;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Map;
+
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.directory.Session;
+import org.nuxeo.ecm.directory.api.DirectoryService;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.model.DefaultComponent;
+
+/**
+ * OAuth2 Client registry component
+ *
+ * @author <a href="mailto:ak@nuxeo.com">Arnaud Kervern</a>
+ * @since 5.9.2
+ */
+public class OAuth2ClientServiceImpl extends DefaultComponent implements OAuth2ClientService {
+
+    @Override
+    public boolean hasClient(String clientId) {
+        OAuth2Client client = getClient(clientId);
+        return client != null && client.isEnabled();
+    }
+
+    @Override
+    public boolean isValidClient(String clientId, String clientSecret) {
+        OAuth2Client client = getClient(clientId);
+        return client != null && client.isValidWith(clientId, clientSecret);
+    }
+
+    @Override
+    public OAuth2Client getClient(String clientId) {
+        DocumentModel doc = getClientModel(clientId);
+        if (doc == null) {
+            return null;
+        }
+        return OAuth2Client.fromDocumentModel(doc);
```

Comment:
Detail: for simple expressions like that I prefer `return doc == null ? null : OAuth2Client.fromDocumentModel(doc)`